parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -1011,8 +1011,10 @@ public <D> CloseableIterable<D> build() {
             conf.unset(property);
           }
           optionsBuilder = HadoopReadOptions.builder(conf);
+          optionsBuilder.withCodecFactory(new ParquetCodecFactory(conf, 0));
         } else {
           optionsBuilder = ParquetReadOptions.builder();
+          optionsBuilder.withCodecFactory(new ParquetCodecFactory(new Configuration(), 0));
```

Comment:
Is it safe to pass `0` for page size? How is that option used?

Reply:
The page size isn't used for decompressors according to [this comment](https://github.com/apache/parquet-mr/blob/6add62754b3d53e30376360f8d215da004fa8096/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/CodecFactory.java#L57). This is what is [currently](https://github.com/apache/parquet-mr/blob/6add62754b3d53e30376360f8d215da004fa8096/parquet-hadoop/src/main/java/org/apache/parquet/ParquetReadOptions.java#L174) being set by default in the options builder.

Reply:
I added comments to make that more clear

Reply:
Idea: Maybe making this a named constant, like `UNUSED_PARQUET_DECOMPRESSOR_PAGE_SIZE` or something would be a good way of indicating that?

Given this solution is temporary, I don’t have a strong feeling either way, but I do like avoiding magic numbers.

Maybe a link to the upstream parquet-mr PR or an Iceberg issue related to zstd decompression would be more informative than just mentioning that page size is essentially ignored?